src/utils/utils.ts

```diff
@@ -24,40 +24,46 @@ export function normalizeDate(dateSrc: string): string | null {
   return dateNorm.toISOString()
 }
 
-export const apiFetchCall = (
-  fetchCleaner: (value: any) => any,
-  url: string,
-  providerName: string
-) => async (): Promise<any | null> => {
-  const options = {
-    method: 'GET',
-    headers: {
-      'Content-Type': 'application/json',
-      Accept: 'application/json'
+export const apiFetchCall =
+  (fetchCleaner: (value: any) => any, url: string, providerName: string) =>
+  async (): Promise<any | null> => {
+    const options = {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json'
+      }
     }
-  }
-  try {
-    const result = await fetch(url, options)
-    if (result.ok === false) {
-      mylog(`${providerName} returned code ${JSON.stringify(result.status)}`)
+    try {
+      const result = await fetch(url, options)
+      if (result.ok === false) {
+        mylog(`${providerName} returned code ${JSON.stringify(result.status)}`)
+      }
+      const jsonObj = await result.json()
+      fetchCleaner(jsonObj)
+      return jsonObj
+    } catch (e) {
+      mylog('Error is:', e)
+      return null
     }
-    const jsonObj = await result.json()
-    fetchCleaner(jsonObj)
-    return jsonObj
-  } catch (e) {
-    mylog('Error is:', e)
-    return null
   }
-}
 
 export const createDatesArray = (
   firstDate: Date,
-  secondDate: Date
+  secondDate: Date,
+  intervalMinutes: number = config.timeBetweenCyclesInMinutes
 ): String[] => {
-  const intervalMS = config.timeBetweenCyclesInMinutes * 60 * 1000
+  if (!(firstDate instanceof Date) || !(secondDate instanceof Date)) {
+    console.log('Date format error')
+    return []
+  }
```

Comment:
Based on your comment, this check seems incomplete. Perhaps it can be combined with line 63?